### ISSUE TITLE
KEYB makes SvarDOS extremely slow on an IBM PC XT (86box)

### ISSUE THREAD

**Issue by mateuszviste** [SvarDOS/bugz#114]
I installed SvarDOS build 20240823 on an emulated IBM PC XT. Have chosen the Polish language. Once everything is installed, the system is extremely slow. Unusable, really - it takes about a second to react to every key press, so typing command takes a looong time. Unloading KEYB makes the system reactive again.

Tried the /9 switch to keyb, without any change.

VM configuration:

![image](https://github.com/user-attachments/assets/dac71f5a-3ecd-4377-be45-43024ac42ee6)

**Comment by mateuszviste** [comment]
changing the kernel (using the FreeDOS kernel) does not improve anything, the problem is still the same.

**Comment by boeckmann** [comment]
Shall I forward this to the FreeDOS mailinglist?

**Comment by mateuszviste** [comment]
changing to a different keyb layout (tested french) also does not change anything.
tried the mkeyb alternative - but it's worse: it immediately freezes.

> Shall I forward this to the FreeDOS mailinglist?

sure :) KEYB hasn't been touched for years, though... Not sure anyone that was involved in it is still around.

**Comment by boeckmann** [comment]
Not the only one :) https://github.com/FDOS/keyb/issues/2

**Comment by mateuszviste** [comment]
this is a known issue, actually. reported 2 years ago: https://github.com/FDOS/keyb/issues/2

**Comment by boeckmann** [comment]
Seems to be fixed: https://gitlab.com/FreeDOS/base/keyb/-/issues/1

**Comment by mateuszviste** [comment]
I saw it, too. And tested this build: https://gitlab.com/FreeDOS/base/keyb/-/tree/master/BIN?ref_type=heads
Keyboard is not lagging anymore, but special keys (ą, ę, ć etc) do not work. And the key "o" becomes mute for some reason. And after a minute or two the system hangs.

**Comment by mateuszviste** [comment]
the gitlab version I've tested is the same as this one https://www.ibiblio.org/pub/micro/pc-stuff/freedos/files/dos/keyb/fdkeyb-2.11/ (v2.11 in file, but the binary still advertises itself as 2.10)

**Comment by boeckmann** [comment]
Ok I will then try to reach the author tomorrow. Can you upload your current image here?

**Comment by mateuszviste** [comment]
Of course. Here is a floppy image I prepared for the occasion: [disk720k.zip](https://github.com/user-attachments/files/16734399/disk720k.zip)

It has the latest KEYB, but the special PL keys do not work on the "IBM PC XT 1982" emulated by 86box (for example ALT+a should generate "ą"). Also the "o" key does not work.

Thanks for your help!

**Comment by boeckmann** [comment]
Can confirm that the keys are "dead". However, have not managed yet to get the system unresponsive. Tried to verify with https://www.pcjs.org/machines/pcx86/ibm/5160/vga/. But this emulator does not support 720K disks for the IBM-XT machine type :(

**Comment by boeckmann** [comment]
German keyboard layout is also slightly broken. The keys requiring an ALT, like the AT sign do not work on the XT machine.

**Comment by boeckmann** [comment]
Bug report filed upstream: https://gitlab.com/FreeDOS/base/keyb/-/issues/12

**Comment by mateuszviste** [comment]
I did not manage to make the system freeze again with keyb 2.11. I did a couple of different tests with different switches yesterday, so maybe the time it hanged it was my fault. But the ALT combos definitely do not work. And what's worse is that the "o" key (without any modifiers) is also dead for some reason, at least with the PL layout. I noticed it when trying to edit autoexec.bat.

**Comment by luzemario** [comment]
For your information, as soon as I run KEYB command on MS-DOS inside QEMU-KVM, MS-DOS crashes. Maybe FreeDOS KEYB uses the same interrupts.
